static/skin/index.js

```diff
@@ -142,10 +142,19 @@
         }
     }
 
+    async function getMagnetLink(downloadLink) {
+        const magnetUrl = downloadLink + '.magnet';
+        const magnetLink = await fetch(magnetUrl).then(response => {
+            return response.ok ? response.text() : '';
+        }).catch((error) => console.log(error));
```

Comment:
Please [add a short timeout](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal#aborting_a_fetch_operation_with_a_timeout) to the `fetch()`. An alternative would be to display an animated progress placeholder for the magnet link in the immediately opened download box, but that is probably too much to ask.

Reply:
What will be the appropriate timeout then?
I feel 3 seconds should be fine

Ping: @kelson42

Reply:
@juuz0 This is maybe a be short, put 10s.